### ISSUE TITLE
CommandBar is not responding to any Click Events/Commands

### ISSUE THREAD

**Issue by joesfshin13** [Microsoft/EasyRepro#740]
Out of the box project all tests that interact with the CommandBar are failing. Especially anything with "New" - Any help would be appreciated this is on a New D365 Implementation from scratch mostly Out of The Box functionality so far.

**Comment by joesfshin13** [comment]
I have tried RelatedGrid as well but seem I still get a dreaded null exception error. Coming from the browserCommand . Any help appreciated or suggestions

**Comment by AngelRodriguez8008** [comment]
Maybe are you running the "Classic Web (XrmBrowser)" tests in an UCI environment?

**Comment by joesfshin13** [comment]
Nope - just seems not to find the element

**Comment by joesfshin13** [comment]
looking for any or possible recommendations

**Comment by AngelRodriguez8008** [comment]
Paste one test that you run and the error please.

**Comment by joesfshin13** [comment]
![1](https://user-images.githubusercontent.com/60446533/73979723-e5e92e00-48eb-11ea-820c-59a47b2bdfae.png)
![2](https://user-images.githubusercontent.com/60446533/73979761-01eccf80-48ec-11ea-8c29-997e34d5f80e.png)

**Comment by joesfshin13** [comment]
all other tests seem to be running fine its just related to the CommadBar tests - New, delete... etc all get the same error message

**Comment by AngelRodriguez8008** [comment]
Sorry I can not reproduce the error this tests are working fine for me. Try the latest version of 'develop' branch.

Thanks & br

**Comment by joesfshin13** [comment]
Follow up question to my issue - could 8x8 integration be causing the issues without being able to click on "New" from CommandBar

**Comment by AngelRodriguez8008** [comment]
I didn't know what "8x8 integration" is or does. I guest is this one ["8x8 integration"](https://www.8x8.com/products/integrations/microsoft-dynamics-365-integration). If this tool change the standard structure of the html, yes it is posible that it is the cause. 

to check that Open the developer pane (F12) search in Elements (Ctrl+F) for the follow XPaths:
```yaml
1. You should be able to find one the Ribbon:
    (a) In the Form: //ul[contains(@data-lp-id,'commandbar-Form')]
or:
    (b) In the Views:  //ul[contains(@data-lp-id,'commandbar-HomePageGrid')]
or:
    (c) you get the error "Unable to find the ribbon." in your EasyRepro tests;

2. And in the next step try to find the command inside the ribbon:
    (a) In the Form: //ul[contains(@data-lp-id,'commandbar-Form')]//li[@aria-label='New']
or:
    (b) In the Views: //ul[contains(@data-lp-id,'commandbar-HomePageGrid')]//li[@aria-label='New']

try also to replace 'New' with other command names.
```
The result should look like:
![image](https://user-images.githubusercontent.com/27274469/75200952-74bcce00-5767-11ea-857a-117d54516688.png)

let me know what you find.

br

**Comment by joesfshin13** [comment]
here is what I see - I am literally been banging my head abouts trying to resolve this its more a goal now than a bug for me now =). I have tried my debugging brains stepping, custom methods to debug the whole boat - I NEED to find this solution just so I can sleep at night but logically I am stumped - 

<button aria-label="New" aria-hidden="true" tabindex="-1" data-id="lead|NoRelationship|HomePageGrid|Mscrm.HomepageGrid.lead.NewRecord" data-lp-id="HomePageGrid:lead-lead|NoRelationship|HomePageGrid|Mscrm.HomepageGrid.lead.NewRecord" type="button" class="u kk gt ea  t kq s kr ks flexbox"><span class="cg ca a fk "><span class="fx  fs kt dh cj a s "><img src="/uclient/resources/images/Add.svg?v=1.3.425-2001.3" alt="" title="" class="aw dk cj dg "></span><span aria-label="New" tabindex="-1" class="fs cj kt ai e am ku do ">New</span></span></button>

**Comment by joesfshin13** [comment]
button aria-label="New" aria-hidden="true" tabindex="-1" data-id="lead|NoRelationship|HomePageGrid|Mscrm.HomepageGrid.lead.NewRecord" data-lp-id="HomePageGrid:lead-lead|NoRelationship|HomePageGrid|Mscrm.HomepageGrid.lead.NewRecord" type="button" class="u kk gt ea  t kq s kr ks flexbox"><span class="cg ca a fk "><span class="fx  fs kt dh cj a s "><img src="/uclient/resources/images/Add.svg?v=1.3.425-2001.3" alt="" title="" class="aw dk cj dg "></span><span aria-label="New" tabindex="-1" class="fs cj kt ai e am ku do ">New</span></span

**Comment by AngelRodriguez8008** [comment]
It is fine that you find the button.  

I set numbers to the XPaths in my comment:
- Did you find it manually or you use the XPath **2. (b)**?
 
I ask because the XPath should find an `<li>` tag, not a button.

Please don't look for the button manually, paste the XPath in the search box, like in the picture show. that simulate what  the EasyRepro code does. It should highlight the HTML element in yellow.
And look in Opportunities view, like the test that you posted shown.  Also look for 1. (b).

I want to know if the HTML structure was changed by 8x8. 

Please also post a picture like that; 
![image](https://user-images.githubusercontent.com/27274469/75365240-30434680-58bd-11ea-8319-8f74caddc121.png)
To check if we have the same App & version.

br

**Comment by joesfshin13** [comment]
![XapthNew](https://user-images.githubusercontent.com/60446533/75371926-da52ac80-5884-11ea-8154-6bcd124ff87b.png)

**Comment by joesfshin13** [comment]
![Clientversion](https://user-images.githubusercontent.com/60446533/75372527-e5f2a300-5885-11ea-9fc5-80613b26efa9.png)

**Comment by AngelRodriguez8008** [comment]
Do not enter your own path, try what I give you.
![image](https://user-images.githubusercontent.com/27274469/75386559-33046280-58e2-11ea-8f2c-3d42347a9c85.png)

**Comment by AngelRodriguez8008** [comment]
![image](https://user-images.githubusercontent.com/27274469/75387926-5b8d5c00-58e4-11ea-87a1-9ae9f931becd.png)
Is it UCI ? In Test you are navigating to UCI > "Sales Hub"

**Comment by joesfshin13** [comment]
Yes I am navigating to UCI - Also plugging in your Xpath examples it works for Views not Forms it finds the element

**Comment by joesfshin13** [comment]
![SalesHubUCI](https://user-images.githubusercontent.com/60446533/75480862-430e5780-595f-11ea-8f00-ba57c83c31f9.png)

**Comment by AngelRodriguez8008** [comment]
Hi, I really want to help you, but I have not enough information. Try follow:
```yaml
In the Sales Hub > Sales > Leads
   1. Click on 'New' & in the Form, search for the XPath:
            //ul[contains(@data-lp-id,'commandbar-Form')]
   That get the whole Ribbon HTML.
   Choose Right click and Copy Element.  
   Paste the Result in a Text File & attach this here.
And:
   2. go back to the Leads View, search for the XPath:  
              //ul[contains(@data-lp-id,'commandbar-HomePageGrid')]
   That get the whole Ribbon HTML.
   Choose Right click and Copy Element.  
   Paste the Result in a Text File & attach this here.
```
Step 1: Looks like:
![image](https://user-images.githubusercontent.com/27274469/75485841-b8077e80-59ab-11ea-84c3-cc7aed237529.png)

Step 2: Looks like:
![image](https://user-images.githubusercontent.com/27274469/75485696-6f4fc580-59ab-11ea-83f1-3c074b259490.png)

**Comment by joesfshin13** [comment]
Ok performed this for you please let me know what I can perform to help better troubleshoot! Thank you again
[HTML from CRM Page.txt](https://github.com/microsoft/EasyRepro/files/4307416/HTML.from.CRM.Page.txt)

**Comment by AngelRodriguez8008** [comment]
From your 'txt' I deduce that in Forms should EasyRepro works fine but, in Views there is no command in the command bar or the HTML was changed in some way.

**Comment by joesfshin13** [comment]
not that I was told - but is there a way I could alter the code on Easyrepro that you see to make this work?

**Comment by AngelRodriguez8008** [comment]
Yes, of course. [Here is the method](https://github.com/microsoft/EasyRepro/blob/af9605bf86f0605e1f37083fa3770006d89f4756/Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs#L1070)
